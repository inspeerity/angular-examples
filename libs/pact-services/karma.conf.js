const { join, resolve } = require("path");
const getBaseKarmaConfig = require("../../karma.conf");
const ports = require("./pact-ports");

module.exports = function(config) {
  const baseConfig = getBaseKarmaConfig();
  config.set({
    ...baseConfig,
    frameworks: [...baseConfig.frameworks, "pact"],
    plugins: [...baseConfig.plugins, require("@pact-foundation/karma-pact")],
    coverageIstanbulReporter: {
      ...baseConfig.coverageIstanbulReporter,
      dir: join(__dirname, "../../coverage/lcov-report/libs/pact-services")
    },
    proxies: {
      "/api/subscription": `http://127.0.0.1:${ports.subscription}/subscription`
    },
    port: 9877,
    spec: 2,
    logLevel: "DEBUG",
    log: resolve(process.cwd(), "logs", "pact.log"),
    pact: [
      {
        port: ports.subscription,
        consumer: "inspeerity",
        provider: "inspeerity-core-subscription",
        dir: resolve(process.cwd(), "pacts")
      }
    ]
  });
};
