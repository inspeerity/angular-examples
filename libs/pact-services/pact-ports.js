module.exports = {
  subscription: 4323
};
