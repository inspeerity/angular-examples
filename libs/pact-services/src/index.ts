export * from "./lib/pact-services.module";
export * from "./lib/subscription";
