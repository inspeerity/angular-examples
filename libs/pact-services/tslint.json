{
  "extends": "../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "ins", "camelCase"],
    "component-selector": [true, "element", "ins", "kebab-case"],
    "nx-enforce-module-boundaries": [false]
  }
}
