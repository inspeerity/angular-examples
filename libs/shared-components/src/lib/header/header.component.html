<div class="header">
  <div fxFlex fxLayout>
    <div>
      <a routerLink="/"> <ins-logo></ins-logo> </a>
    </div>
    <mat-toolbar color="primary">
      <div
        fxFlex
        fxLayout
        fxLayoutAlign="space-between center"
        class="header__container"
      >
        <div class="header__name" fxHide.xs>
          <div
            *ngIf="isAuthenticated"
            [innerHTML]="'HEADER.WELCOME' | translate: { userName: userName }"
          ></div>
        </div>
        <ul
          fxLayout
          fxLayoutGap="2rem"
          fxLayoutAlign="center center"
          class="header__items"
        >
          <li fxHide.xs *ngIf="!isAuthenticated">
            <a class="header__link"
               routerLink="/sign-in"
               [translate]="'HEADER.NAVIGATION.SIGN_IN'"
            ></a>
          </li>
          <li *ngIf="!isAuthenticated">
            <a mat-stroked-button routerLink="/sign-up">
              <span [translate]="'HEADER.NAVIGATION.GET_STARTED'"></span>
            </a>
          </li>
          <li *ngIf="isAuthenticated"">
            <a class="header__link"
               (click)="logout()"
               [translate]="'HEADER.NAVIGATION.SIGN_OUT'"
            ></a>
          </li>
        </ul>
      </div>
      <div fxHide.gt-xs>
        <button mat-icon-button (click)="toggleSidenav()">
          <mat-icon>menu</mat-icon>
        </button>
      </div>
    </mat-toolbar>
  </div>
</div>
