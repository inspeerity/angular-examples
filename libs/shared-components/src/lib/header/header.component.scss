@import "~sass/abstracts/mixins";
@import "~sass/abstracts/placeholders";
@import "~sass/abstracts/variables";
@import "~sass/theme/variables";

.header {
  position: relative;
  @extend %grid-container;
  @include respond(phone) {
    margin: 0 2rem;
  }

  &__items {
    font-size: 1.8rem;
    line-height: 2.2rem;
  }

  &__link {
    font-size: 1.8rem;
    line-height: 2.3rem;
    color: $brand-color-black;
    text-decoration: none;
  }
}
