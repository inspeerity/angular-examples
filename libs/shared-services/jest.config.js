module.exports = {
  name: 'shared-services',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/lcov-report/libs/shared-services'
};
