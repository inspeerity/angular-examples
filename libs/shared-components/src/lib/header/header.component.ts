import { Component, EventEmitter, OnInit, Output } from "@angular/core";

import { AuthenticationService } from "@inspeerity/pact-services";
import { TokenService } from "@inspeerity/shared-services";

@Component({
  selector: "ins-header",
  templateUrl: "./header.component.html",
  styleUrls: ["./header.component.scss"]
})
export class HeaderComponent implements OnInit {
  @Output() sidenavToggleEvent = new EventEmitter<void>();

  isAuthenticated: boolean;
  userName: string;

  constructor(
    private authenticationService: AuthenticationService,
    private tokenService: TokenService
  ) {}

  ngOnInit() {
    this.authenticationService.authenticationState.subscribe(
      isAuthenticated => {
        this.isAuthenticated = isAuthenticated;
        this.userName = this.tokenService.userName;
      }
    );
  }

  toggleSidenav() {
    this.sidenavToggleEvent.emit();
  }

  logout() {
    this.authenticationService.logout();
  }
}
