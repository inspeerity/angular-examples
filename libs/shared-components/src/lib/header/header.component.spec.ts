import { of } from "rxjs";

import {
  createComponentFixture,
  provideMagicalMock
} from "@inspeerity/tools/testing";

import { AuthenticationService } from "@inspeerity/pact-services";
import { TokenService } from "@inspeerity/shared-services";

import { HeaderComponent } from "./header.component";

describe("HeaderComponent", () => {
  const authenticationServiceValue = {
    authenticationState: of(true),
    logout: jest.fn()
  };
  const fixture = createComponentFixture({
    component: HeaderComponent,
    providers: [
      {
        provide: AuthenticationService,
        useValue: authenticationServiceValue
      },
      provideMagicalMock(TokenService)
    ]
  });
  let authenticationService: AuthenticationService, tokenService: TokenService;

  beforeEach(async () => {
    await fixture.compile();
  });

  beforeEach(() => {
    authenticationService = fixture.get(AuthenticationService);
    tokenService = fixture.get(TokenService);
  });

  it("should create", () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it("toggleSidenav() should emit a new value", () => {
    jest.spyOn(fixture.componentInstance.sidenavToggleEvent, "emit");

    fixture.componentInstance.toggleSidenav();

    expect(
      fixture.componentInstance.sidenavToggleEvent.emit
    ).toHaveBeenCalled();
  });

  it("should call logout method", () => {
    fixture.componentInstance.logout();

    expect(authenticationService.logout).toHaveBeenCalled();
  });
});
