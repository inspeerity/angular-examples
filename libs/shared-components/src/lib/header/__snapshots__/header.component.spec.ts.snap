// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should create 1`] = `
<div
  id="root0"
  ng-version="6.1.10"
>
  <div
    class="header"
  >
    <div
      fxflex=""
      fxlayout=""
    >
      <div>
        <a
          routerlink="/"
        >
          <inspeerity-logo />
        </a>
      </div>
      <mat-toolbar
        color="primary"
      >
        <div
          class="header__container"
          fxflex=""
          fxlayout=""
          fxlayoutalign="space-between center"
        >
          <div
            class="header__name"
          >
            <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
            <div>
              HEADER.WELCOME
            </div>
          </div>
          <ul
            class="header__items"
            fxlayout=""
            fxlayoutalign="center center"
            fxlayoutgap="2rem"
          >
            <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
            <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
            <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
            <li
            >
              <a
                ng-reflect-translate="HEADER.NAVIGATION.SIGN_OUT"
              >
                HEADER.NAVIGATION.SIGN_OUT
              </a>
            </li>
          </ul>
        </div>
        <div
          fxhide.gt-xs=""
        >
          <button
            mat-icon-button=""
          >
            <mat-icon>
              menu
            </mat-icon>
          </button>
        </div>
      </mat-toolbar>
    </div>
  </div>
</div>
`;
